lib/SILGen/SILGenBuiltin.cpp

```diff
@@ -1042,28 +1042,60 @@ static ManagedValue emitBuiltinTypeTrait(SILGenFunction &SGF,
 }
 
 // SWIFT_ENABLE_TENSORFLOW
-static ManagedValue emitBuiltinAutoDiffGetJVP(SILGenFunction &SGF,
-                                              SILLocation loc,
-                                              SubstitutionMap substitutions,
-                                              Expr *argument,
-                                              SGFContext C) {
-  auto argVal = SGF.emitRValue(argument);
-  auto jvp = SGF.getBuilder().createAutoDiffFunctionExtract(
-      loc, AutoDiffFunctionExtractee::JVP, /*differentiationOrder*/ 1,
-      std::move(argVal).forwardAsSingleValue(SGF, loc));
-  return ManagedValue::forUnmanaged(jvp);
+static ManagedValue emitBuiltinAutoDiffApplyAssociatedFunction(
+    AutoDiffAssociatedFunctionKind kind, SILGenFunction &SGF, SILLocation loc,
+    SubstitutionMap substitutions, Expr *argument, SGFContext C) {
+  auto *argTuple = cast<TupleExpr>(argument);
+  assert(argTuple->getNumElements() == 2);
+  auto origFnVal =
+      SGF.emitRValueAsSingleValue(argTuple->getElement(0)).getUnmanagedValue();
+  auto origFnArgVal =
+      SGF.emitRValueAsSingleValue(argTuple->getElement(1)).forward(SGF);
```

Comment:
Oh, I wanted to comment on one more thing.

I think you need to `.borrow(SGF).forward(SGF)` this value when you're passing it in as a `@guaranteed`/`@in_guaranteed` parameter. `.forward(SGF)` is for when you're passing it in as a `@owned`/`@in` parameter.

Most parameters seem guaranteed nowaways, so `.borrow(SGF).forward(SGF)` will usually be right, and `.forward(SGF)` will usually leak memory because it stops the value from eventually getting destroyed. But the most right think is to check the parameter convention and do the right thing.

I'm not 100% sure about this, so maybe you should just add a TODO and then someone can experiment with it or ask about it before doing it.

Reply:
Called `borrow`. It worked fine.